charts/k8s-monitoring/templates/alloy-servicegraph.yaml

```
@@ -0,0 +1,64 @@
+{{- range $destination := .Values.destinations }}
+{{- if eq $destination.type "otlp" }}
+  {{- $isServiceGraphsEnabled := include "destinations.otlp.isServiceGraphsEnabled" $destination }}
+  {{- if eq $isServiceGraphsEnabled "true" }}
+
+    {{- $defaultValues := "destinations/otlp-values.yaml" | $.Files.Get | fromYaml }}
+    {{- $destination = mergeOverwrite $defaultValues $destination }}
+    {{- $metricsDestinationNames := include "destinations.get" (dict "destinations" $.Values.destinations "type" "metrics" "ecosystem" "otlp" "filter" $destination.processors.serviceGraphMetrics.destinations) | fromYamlArray -}}
+
+    {{- $destinations := list }}
+    {{- $destinationTargets := list }}
+    {{- range $metricsDestinationName := $metricsDestinationNames }}
+      {{- $myDestination := include "destination.getDestinationByName" (deepCopy $ | merge (dict "destination" $metricsDestinationName )) | fromYaml }}
+      {{- $destinations = append $destinations $myDestination }}
+      {{- if ne $myDestination.type "prometheus" -}}
+        {{- $destinationTarget := include "destinations.otlp.alloy.exporter.target" $myDestination }}
+        {{- $destinationTargets = append $destinationTargets (printf "%s" $destinationTarget) }}
+      {{- else }}
+        {{- $destinationTarget := include "destinations.prometheus.alloy.otlp.metrics.target" $myDestination }}
+        {{- $destinationTargets = append $destinationTargets (printf "%s" $destinationTarget) }}
+      {{- end }}
+    {{- end }}
+
+    {{- $serviceGraphConfig := merge (dict "traces" $destination "metrics" $destinationTargets "name" "service_graph_metrics") $destination.processors.serviceGraphMetrics }}
+    {{- $serviceGraphTarget := include "servicegraph.connector.serviceGraphMetrics.alloy.target" $serviceGraphConfig }}
+    {{- $receiverConfig := merge (dict "traces" $serviceGraphTarget "name" "service_graph_receiver") $destination.processors.serviceGraphMetrics }}
+
+    {{- $alloyConfig := "" }}
+    {{- $alloyConfig = cat $alloyConfig ((include "receiver.otlp.alloy" $receiverConfig) | trim | nindent 0) }}
```

This might remove the `config: |2` thing we see when it's set.
```suggestion
    {{- $alloyConfig = cat $alloyConfig ((include "receiver.otlp.alloy" $receiverConfig) | trim | indent 0) }}
```